cpplite/cpplite.debugger/test/unit/src/org/netbeans/modules/cpplite/debugger/ProcessParametersTest.java

```diff
@@ -0,0 +1,192 @@
+            "   if (getcwd(buffer, sizeof(buffer)) != NULL) {\n" +
+            "       printf(\"CWD:%s\\n\", buffer);\n" +
+            "   } else {\n" +
+            "       perror(\"getcwd() error\");\n" +
+            "       return 1;\n" +
+            "   }\n" +
+            "   while (--argc) {\n" +
+            "      printf(\"ARG:%s\\n\", *(++argv));\n" +
+            "   }\n" +
+            "   for (char **e = envp; *e; e++) {\n" +
+            "      printf(\"ENV:%s\\n\", *e);\n" +
+            "   }\n" +
+            "   return 0;\n" +
+            "}";
+
+    public ProcessParametersTest(String s) {
+        super(s);
+    }
+
+    @Override
+    protected void setUp() throws Exception {
+        clearWorkDir();
+    }
+
+    public void testWorkingDir() throws Exception {
+        File wd = getWorkDir();
+        createSourceFile("main.cpp", wd, CPP_APP);
+        compileCPP("main", wd);
+        File appWD = Files.createTempDirectory("TestCWD").toFile();
+        try {
+            ExplicitProcessParameters params = ExplicitProcessParameters.builder().
+                    workingDirectory(appWD).
+                    build();
+            startDebugging("main", wd, params);
+            assertEquals(0, waitAppProcessExit());
+            String output = stdOut.toString();
+            assertEquals("CWD:" + appWD.getAbsolutePath(), output.substring(0, output.indexOf('\n')));
+        } finally {
+            appWD.delete();
+        }
+    }
+
+    public void testArgumentsOwn() throws Exception {
+        File wd = getWorkDir();
+        createSourceFile("main.cpp", wd, CPP_APP);
+        compileCPP("main", wd);
+        ExplicitProcessParameters params = ExplicitProcessParameters.empty();
+        startDebugging(Arrays.asList(new File(wd, "main").getAbsolutePath(), "Param1", "Param 2"), params);
+        assertEquals(0, waitAppProcessExit());
+        assertArguments("Param1", "Param 2");
+    }
+
+    public void testArgumentsExplicit() throws Exception {
+        File wd = getWorkDir();
+        createSourceFile("main.cpp", wd, CPP_APP);
+        compileCPP("main", wd);
+        ExplicitProcessParameters params = ExplicitProcessParameters.builder().
+                arg("ExpParam1").
+                arg("ExpParam 2").
+                build();
+        startDebugging(Arrays.asList(new File(wd, "main").getAbsolutePath()), params);
+        assertEquals(0, waitAppProcessExit());
+        assertArguments("ExpParam1", "ExpParam 2");
+    }
+
+    public void testArgumentsCombinedReplace() throws Exception {
+        File wd = getWorkDir();
+        createSourceFile("main.cpp", wd, CPP_APP);
+        compileCPP("main", wd);
+        ExplicitProcessParameters params = ExplicitProcessParameters.builder().
+                arg("ExpParam1").
+                arg("ExpParam 2").
+                build();
+        startDebugging(Arrays.asList(new File(wd, "main").getAbsolutePath(), "Param1", "Param 2"), params);
```

Comment:
combined, explicit, own parameters - that looks like explicit set of tests. Thanks.